src/sentry/api/serializers/rest_framework/notification_action.py

```diff
@@ -213,6 +214,7 @@ def validate_slack_channel(
         # If we've received a channel and id, verify them against one another
         if channel_name and channel_id:
             try:
+                assert self.integration, "Associated integration must exist to validate channel id"
```

Comment:
I suspect that integration should never be None -- should this validation happen when integration is assigned above?  I suspect this is an oversight from the HC migration (where it would have always been non-None from the datastore)

Reply:
Good pt 👍